src/compiler/checker.ts

```diff
@@ -26093,6 +26093,9 @@ namespace ts {
                     if (i !== (parameterCount - 1)) {
                         return grammarErrorOnNode(parameter.dotDotDotToken, Diagnostics.A_rest_parameter_must_be_last_in_a_parameter_list);
                     }
+                    if (parameters.hasTrailingComma) {
+                        return grammarErrorOnNode(parameter.dotDotDotToken, Diagnostics.A_rest_parameter_may_not_have_a_trailing_comma);

```

Comment:
I would say that this is the wrong span to error on. Either error on the commma, or the parameter itself.